version: "3.8"

services:
  database:
    image: postgres:16-alpine
    environment:
      POSTGRES_DB: ${POSTGRES_DB}
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
    ports:
      - "5432:5432"
    healthcheck:
      test: [ "CMD-SHELL", "sh -c 'pg_isready -U ${POSTGRES_USER} -d ${POSTGRES_DB}'" ]
      interval: 10s
      timeout: 3s
      retries: 3
  backend:
    build: ./backend
    command: >
      sh -c "python src/manage.py migrate &&
             python src/manage.py loaddata initial_data.json &&
             python src/manage.py runserver 0.0.0.0:8000"
    ports:
      - "8000:8000"
    depends_on:
      database:
        condition: service_healthy
    environment:
      DB_NAME: ${POSTGRES_DB}
      DB_USER: ${POSTGRES_USER}
      DB_PASSWORD: ${POSTGRES_PASSWORD}
      DB_HOST: database
      DB_PORT: 5432
      CORS_ALLOWED_ORIGINS: ${APP_ADDRESS}:3000
    volumes:
      - ./backend/src:/app/src
  frontend:
    build: ./frontend
    ports:
      - "3000:3000"
    depends_on:
      - backend
    environment:
      VITE_APP_API_URL: ${APP_ADDRESS}:8000
    volumes:
      - ./frontend/src:/app/src